task/github.py

```diff
@@ -296,42 +299,44 @@ def delete(self, resource, accept=[]):
         else:
             return None
 
-    def patch(self, resource, data, accept=[]):
+    def patch(self, resource: str, data: str, accept: List[Any] = []) -> Any:
         response = self.request("PATCH", resource, json.dumps(data), {"Content-Type": "application/json"})
         status = response['status']
         if (status < 200 or status >= 300) and status not in accept:
             raise GitHubError(self.qualify(resource), response)
         self.cache.mark()
         return json.loads(response['data'])
 
-    def statuses(self, revision):
+    def statuses(self, revision: str) -> Dict[Any, Any]:
         result = {}
         page = 1
         count = 100
         while count == 100:
             data = self.get("commits/{0}/status?page={1}&per_page={2}".format(revision, page, count))
-            count = 0
-            page += 1
-            if "statuses" in data:
-                for status in data["statuses"]:
-                    if is_valid_context(status["context"], self.repo) and status["context"] not in result:
-                        result[status["context"]] = status
-                count = len(data["statuses"])
+            if data is not None:
```

Comment:
Shouldn't this be more like an assertion? If we get a None data in the middle of the iteration, this is either a broken GitHhub API or some broken logic on our side. But we really should never get None here, at most an empty list?

Reply:
Seems I made a bogus change though with `if data and "statuses" in data:`

Reply:
Ok, the handling if `None` is required if Github returns 404, if that's a fatal error, then we should assert yeah.